SecondHomework/Logistic/ParcelManager.cs

```
@@ -1,134 +1,48 @@
-
-        AddParcel(name, recipient, destination);
-    }
-
-    public void Display()
+    public void AddParcel(Parcel parcel)
     {
-        if (parcels.Count == 0)
-        {
-            Logger.NoParcels();
-        }
-        else
-        {
-            foreach (var parcel in parcels)
-            {
-                Console.WriteLine(parcel);
-            }
-        }
-    }
-
-    public void AddParcel(string? name, string? recipient, string? destination)
-    {
-        var parcel = new Parcel(Guid.NewGuid(), name, recipient, destination);
         parcels.Add(parcel);
-        SaveParcels();
+        fileManager.Save(parcels);
     }
 
-    public void SaveParcels()
+    public List<Parcel> ReadParcels()
     {
-        var json = JsonConvert.SerializeObject(parcels, Newtonsoft.Json.Formatting.Indented);
-        File.WriteAllText(_filePath, json);
+        parcels = fileManager.Read();
+        return parcels;
     }
 
-    public void CheckIfDeleteAndRemove()
+    public bool RemoveParcel(Guid id, List<Parcel> parcelsList, out string message)
     {
-        string? answer = Console.ReadLine();
-        if (answer?.ToLower() == "yes")
+        try
         {
-            Logger.CheckForIdToRemove();
-            string? id = Console.ReadLine();
-            if (Guid.TryParse(id, out Guid Id))
+            Parcel parcelToRemove = parcelsList.Find(p => p.Id == id);
+            if (parcelToRemove != null)
             {
-                DeleteParcels(Id);
-                Console.WriteLine("Updated parcels are:");
-                Display();
+                parcelsList.Remove(parcelToRemove);
+                fileManager.Save(parcelsList);
+                message = $"Parcel with ID: {id} removed sucessfully.";
+                return true;
             }
             else
             {
-                Console.WriteLine("Invalid ID format.");
+                message = $"Parcel with ID: {id} not found.";
```

I would throw exception instead of returning message